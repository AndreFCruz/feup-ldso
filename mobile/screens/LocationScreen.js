import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Image, View, Text } from 'react-native';
import { connect } from 'react-redux';
import { locations } from '../constants/Locations';
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';
import facultyStyles from '../constants/SpecificStyles';

class LocationScreen extends React.Component {

    render() {
        const { name } = this.props;
        const qrcode = 'https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=' + locations[name].qrcode;
        
        return (
            <View style={styles.container}>
                <View style={styles.mapContainer}> 
                    <Image style={styles.map} source={locations[name].uri} />
                </View>

                <View style={styles.rightSide}>
                    <View style={[styles.box, facultyStyles[name].factBackground]} >
                        <View style={[styles.imageContainer, facultyStyles[name].factBackground]}>
                            <Image style={styles.image} source={{uri: qrcode}}/>
                        </View>
                        
                        <Text style={[styles.textContainer, facultyStyles[name].factText]}> 
                            {this.props.language == 'en' ? 'Scan me with a QR code reader and wait for the camera to detect automatically' 
                                : 'Inicie o leitor de códigos QR e espere que a câmara o detete automaticamente'}
                        </Text>
                    </View>
                </View>

                <View style={[styles.textBox, facultyStyles[name].factBackground]}>
                    <Text style={[styles.text, facultyStyles[name].factText]}>
                        {this.props.language == 'en' ? 'Get to know ' 
                            : 'Venha conhecer a '}
                        {name.toUpperCase()}
                    </Text>
                </View>
            </View>
        );
    }
}

LocationScreen.propTypes = {
    name: PropTypes.string,
    language: PropTypes.string,
};

const styles = StyleSheet.create({
    container: { 
        flex: 1,
        flexDirection: 'row'
    } ,

    mapContainer:  {
        flex: 7.5,
    },

    map: {
        flex: 1,
        width: wp('100%'),
        resizeMode: 'cover'      
    },

    box: {
        width: wp('17%'),
        height: hp('42%'),
        borderRadius: hp('2%'),
        justifyContent: 'center',
        alignItems: 'center',
        elevation: 8,
    },

    rightSide: {
        flex: 2.5,
        marginLeft: wp('5%'),
        marginTop: hp('2%'),
        alignItems: 'center'
    },

    imageContainer:  {
        flex: 2,
        justifyContent: 'center',
        alignItems: 'center'
    },

    image: {
        flex: 1,
        resizeMode: 'contain',
        width: wp('12%'),
        
    },

    textBox: {
        position: 'absolute',
        left: wp('2%'),
        top: hp('2%'),
        elevation: 8,
        borderRadius: hp('1%'),
    },

    text: {
        paddingLeft: wp('1.25%'),
        paddingRight: wp('1.25%'),
        paddingBottom: hp('1%'),
        paddingTop: hp('1%'),
        justifyContent: 'center',
        alignItems: 'center',
        fontSize: hp('3%'),
        textAlign: 'center',
        fontFamily: 'Quicksand_regular',
    },

    textContainer: {
        flex: 1,
        paddingLeft: wp('2%'),
        paddingRight: wp('1.5%'),
        paddingBottom: hp('2%'),
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
        fontSize: hp('2.1%'),
        fontFamily: 'OpenSans_regular',
    }

});

const mapStateToProps = ({ faculty, language }) => ({
    name: faculty.name,
    language: language.selection
});

export default connect(mapStateToProps)(LocationScreen);
