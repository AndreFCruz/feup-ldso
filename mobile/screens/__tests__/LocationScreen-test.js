import 'react-native';
import React from 'react';
import LocationScreen from '../LocationScreen';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import { Image, Text } from 'react-native';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

function setup(lang) {

    const initialState = { 
        faculty: {
            name: 'feup',
        },
        language: {
            selection: lang,
        }
    };

    const wrapper = shallow(<LocationScreen />,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: true
        });

    return wrapper;
}

it('renders correct number of elements', () => {
    const wrapper = setup('en');

    expect(wrapper.dive().children().length).toEqual(3);
    expect(wrapper.dive().childAt(0).childAt(0).find(Image).exists()).toEqual(true);
    expect(wrapper.dive().childAt(1).childAt(0).children().length).toEqual(2);
    expect(wrapper.dive().childAt(2).childAt(0).find(Text).exists()).toEqual(true);
});

it('renders expected text', () => {
    const wrapper = setup('en');
    const text = 'Scan me with a QR code reader and wait for the camera to detect automatically';
    
    expect(wrapper.dive().childAt(1).childAt(0).childAt(1).shallow().text()).toEqual(text);
});

it('renders screen placeholder content', () => {
    const wrapper = setup('en');

    expect(wrapper.dive()).toMatchSnapshot();
});

it('renders in other language', () => {
    const wrapper = setup('pt');

    expect(wrapper.dive()).toMatchSnapshot();
});