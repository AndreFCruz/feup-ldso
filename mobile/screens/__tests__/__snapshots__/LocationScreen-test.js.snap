// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders in other language 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Component
    style={
      Object {
        "flex": 7.5,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 1,
          "resizeMode": "cover",
          "width": 750,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2.5,
        "marginLeft": 37.5,
        "marginTop": 26.5,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 26.5,
            "elevation": 8,
            "height": 560.5,
            "justifyContent": "center",
            "width": 127.5,
          },
          Object {
            "backgroundColor": "#9f2d20",
          },
        ]
      }
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 2,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#9f2d20",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=https://goo.gl/maps/Dn8UfgUFN4G2",
            }
          }
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
              "width": 90,
            }
          }
        />
      </Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "fontFamily": "OpenSans_regular",
              "fontSize": 28,
              "justifyContent": "center",
              "paddingBottom": 26.5,
              "paddingLeft": 15,
              "paddingRight": 11.5,
              "textAlign": "center",
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        Inicie o leitor de códigos QR e espere que a câmara o detete automaticamente
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "borderRadius": 13.5,
          "elevation": 8,
          "left": 15,
          "position": "absolute",
          "top": 26.5,
        },
        Object {
          "backgroundColor": "#9f2d20",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignItems": "center",
            "fontFamily": "Quicksand_regular",
            "fontSize": 40,
            "justifyContent": "center",
            "paddingBottom": 13.5,
            "paddingLeft": 9.5,
            "paddingRight": 9.5,
            "paddingTop": 13.5,
            "textAlign": "center",
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Venha conhecer a 
      FEUP
    </Text>
  </Component>
</Component>
`;

exports[`renders screen placeholder content 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Component
    style={
      Object {
        "flex": 7.5,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 1,
          "resizeMode": "cover",
          "width": 750,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2.5,
        "marginLeft": 37.5,
        "marginTop": 26.5,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 26.5,
            "elevation": 8,
            "height": 560.5,
            "justifyContent": "center",
            "width": 127.5,
          },
          Object {
            "backgroundColor": "#9f2d20",
          },
        ]
      }
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 2,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#9f2d20",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=https://goo.gl/maps/Dn8UfgUFN4G2",
            }
          }
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
              "width": 90,
            }
          }
        />
      </Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "fontFamily": "OpenSans_regular",
              "fontSize": 28,
              "justifyContent": "center",
              "paddingBottom": 26.5,
              "paddingLeft": 15,
              "paddingRight": 11.5,
              "textAlign": "center",
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        Scan me with a QR code reader and wait for the camera to detect automatically
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "borderRadius": 13.5,
          "elevation": 8,
          "left": 15,
          "position": "absolute",
          "top": 26.5,
        },
        Object {
          "backgroundColor": "#9f2d20",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignItems": "center",
            "fontFamily": "Quicksand_regular",
            "fontSize": 40,
            "justifyContent": "center",
            "paddingBottom": 13.5,
            "paddingLeft": 9.5,
            "paddingRight": 9.5,
            "paddingTop": 13.5,
            "textAlign": "center",
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Get to know 
      FEUP
    </Text>
  </Component>
</Component>
`;
